query/functions.go

```diff
@@ -2150,3 +2156,110 @@ func (r *UnsignedBottomReducer) Emit() []UnsignedPoint {
 	sort.Sort(sort.Reverse(&h))
 	return points
 }
+
+type StringMergeHllReducer struct {
+	plus *hll.Plus
+	err  error
+}
+
+func NewStringMergeHllReducer() *StringMergeHllReducer {
+	return &StringMergeHllReducer{plus: nil}
+}
+
+func unmarshalPlus(s string) (*hll.Plus, error) {
+	if string(hllPrefix) != s[:len(hllPrefix)] {
+		if string(hllErrorPrefix) == s[:len(hllErrorPrefix)] {
+			// parse a special error out of the string.
+			return nil, fmt.Errorf("%v", s[len(hllErrorPrefix):])
+		}
```

Comment:
error case when unmarshalling

Reply:
How often do `marshalPlus` and `unmarshalPlus` get called compared to the method where the `error` return was removed? I.e., do we know that this performs better than returning an error?

Reply:
I guess this this way impacts `count_hll` but avoids impacting all other aggregates?

Reply:
Yes exactly. I'm not too worried about slowing `count_hll`, especially since it actually can have errors, but I don't want to slow down `sum`, `count`, `mean`, etc.